nomad/node_endpoint.go

```diff
@@ -536,9 +536,18 @@ func (n *Node) UpdateDrain(args *structs.NodeUpdateDrainRequest,
 		}
 	}
 
+	// Mark start time for the drain
+	if args.DrainStrategy != nil {
+		if node.DrainStrategy == nil || node.DrainStrategy.StartedAt.IsZero() {
+			args.DrainStrategy.StartedAt = time.Now().UTC()
+		} else {
+			args.DrainStrategy.StartedAt = node.DrainStrategy.StartedAt
+		}
+	}
+
 	// Mark the deadline time
 	if args.DrainStrategy != nil && args.DrainStrategy.Deadline.Nanoseconds() > 0 {
-		args.DrainStrategy.ForceDeadline = time.Now().Add(args.DrainStrategy.Deadline)
+		args.DrainStrategy.ForceDeadline = time.Now().Add(args.DrainStrategy.Deadline).UTC()
```

Comment:
I like @jrasell's idea of having a var startedAt time that we could then use here as well to add on to. There seems to be at least one more location calling time.Now that could also benefit from using a shared variable